### PR TITLE
test(custom-scheduler): stabilize custom scheduler testing

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description

After implementing the reviewer fragment thing in the new reviewer with #16455 the custom scheduler test was failing for me on the command line so my customary / frequent `./gradlew jacocoTestReport` did not pass

I believe others did not notice it because:

1- it is not executed in CI because of a Flaky annotation
2- I suppose people developing locally either don't run tests or run them from Android Studio? I use the command line

I need that to pass, so I looked into it, and this or something like this will fix it

## Fixes
* Fixes regression from  #16455

## Approach

1- realize by view hierarchy inspection that the hierarchies are different from the command line and from Android Studio. I have no idea. They are though. So...allow either name for the the layout flip or ease 3 button view resource to match 🤷 
2- realize that timing issues are still present (even for me locally) but if we use the otherwise-unused retry rule with a retry of 3, it works every time locally! So I am passing now
3- try to get this working in CI so it doesn't start silently failing again and sending me on a yak shave. Bumping retry to 10 seems to do it? 🙏 

Note there was a line-ending issue in initial push that has been peeled off to #16536 

## How Has This Been Tested?

It's all tests, it is made of tests. But this was a flake, so it has a a higher burden of proof. I'm 9 for 9 running this so I think I actually stabilized it

## Learning (optional, can help others)

- New emulator on new ubuntu runner is pretty quick, default view match timeout is 10s and we don't need 30s anymore
- we already had a retry rule which was cool
- view hierarchies can be different between command line and android studio, and I can't even begin to explain that
- I cannot find a way to run *just* a single test or regex of tests from the androidTest hierarchy via command line 🤷 

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
(note, this passes over and over again on my fork - example - https://github.com/mikehardy/Anki-Android/actions/runs/9341737197/job/25709045233 )

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt [hunk 1]
```diff
@@ -1,176 +1,209 @@
...
+            assertThat(cardFromDb.easeFactor, equalTo(3000))
+            assertThat(cardFromDb.interval, equalTo(123))
+            assertThat(cardFromDb.customData, equalTo("""{"c":2}"""))
+        } catch (e: Exception) {
+            ThreadUtils.sleep(2000)
+            cardFromDb = col.getCard(card.id).toBackendCard()
+            assertThat(cardFromDb.easeFactor, equalTo(3000))
+            assertThat(cardFromDb.interval, equalTo(123))
+            assertThat(cardFromDb.customData, equalTo("""{"c":2}"""))
+        }
```
Comment (david-allison) [inline AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt:105]:
```patch
Index: AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt b/AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt
--- a/AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt	(revision 462cd03d90764434c0eddb4537d462dbb7b79422)
+++ b/AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt	(date 1717412447111)
@@ -91,17 +91,18 @@
 
         clickShowAnswerAndAnswerGood()
 
-        cardFromDb = col.getCard(card.id).toBackendCard()
-        try {
+        fun runAssertion() {
+            cardFromDb = col.getCard(card.id).toBackendCard()
             assertThat(cardFromDb.easeFactor, equalTo(3000))
             assertThat(cardFromDb.interval, equalTo(123))
             assertThat(cardFromDb.customData, equalTo("""{"c":2}"""))
+        }
+        
+        try {
+            runAssertion()
         } catch (e: Exception) {
             ThreadUtils.sleep(2000)
-            cardFromDb = col.getCard(card.id).toBackendCard()
-            assertThat(cardFromDb.easeFactor, equalTo(3000))
-            assertThat(cardFromDb.interval, equalTo(123))
-            assertThat(cardFromDb.customData, equalTo("""{"c":2}"""))
+            runAssertion()
         }
     }
 
```

**Review by david-allison — CHANGES_REQUESTED** [review]
Rejecting due to spacing changes in commit 1 on `ReviewerTest`

Approve after this is fixed, suggestion is a nitpick for clarity

**Comment by mikehardy** [comment]
Yeah the spacing change is ridiculous and will definitely be changed if this passes muster at all:

> If this works for everyone else I'll fix up the ridiculous "full file line-ending" change thing that happened on my computer. No idea, and I don't want to shave that yak until there's buy in on this PR. View diff with spacing changes ignored

Seems like it might - with nitpick for clarity 👍  - so will clean this up a bit and repush

**Review by david-allison — APPROVED** [review]
Fantastic, cheers!

File: AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/androidTest/java/com/ichi2/anki/ReviewerTest.kt]: I also did not like the duplication. Adopted this solution verbatim

**Comment by mikehardy** [comment]
image

**Comment by mikehardy** [comment]
Succeeded first time here as well, and the queue will run it again and then it'll run again again on push to main, so 3 more runs to give it a final look and hope it is actually non-flaky

If this starts to flake again - for anyone reading this in the future - just readd the Flaky annotations, do not change any of the other work here as it was all necessary (even the retries!) in order for it to work locally even.

**Pull request merged**